sympy/utilities/runtests.py

```diff
@@ -530,11 +530,6 @@ def _test(*paths, **kwargs):
                    fast_threshold=fast_threshold,
                    slow_threshold=slow_threshold)
 
-    # Disable warnings for external modules
-    import sympy.external
-    sympy.external.importtools.WARN_OLD_VERSION = False
-    sympy.external.importtools.WARN_NOT_INSTALLED = False
```

Comment:
I think these were put in here because if someone has an old version of say matplotlib installed, it should just run the tests skipping the matplotlib tests. 

What is the behavior after this change? Does it just warn or do these warnings produce exceptions?

Reply:
I think that these settings are there because otherwise various tests would causes warnings to be released. I've moved code to test for those warnings into the tests.

Reply:
The reason I removed those lines is so that the tests work under `bin.test`, `pytest` *and* when imported and run normally: `import test_module; test_module.test_stuff()`. For those three to be equivalent it's best not to have special settings in the test runner.

Reply:
Yes but aren't warnings turned into exceptions in the tests (or maybe that's just DeprecationWarning).

Reply:
Only `DeprecationWarning`s are turned into errors (immediately below the code you're quoting). It looks as if that happens so that `raises` can be used to test for `DepreactionWarning`s.

I could configure `pytest` to also use the same setup as `bin/test` but I think it's better to test with `warns`. Otherwise the tests won't work when run outside the test runner.